samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -54,26 +56,26 @@ class InvokeContext:
 
     def __init__(
         self,  # pylint: disable=R0914
-        template_file,
-        function_identifier=None,
-        env_vars_file=None,
-        docker_volume_basedir=None,
-        docker_network=None,
-        log_file=None,
-        skip_pull_image=None,
-        debug_ports=None,
-        debug_args=None,
-        debugger_path=None,
-        container_env_vars_file=None,
-        parameter_overrides=None,
-        layer_cache_basedir=None,
-        force_image_build=None,
-        aws_region=None,
-        aws_profile=None,
-        warm_container_initialization_mode=None,
-        debug_function=None,
-        shutdown=False,
-    ):
+        template_file: str,
+        function_identifier: Optional[str] = None,
+        env_vars_file: Optional[str] = None,
+        docker_volume_basedir: Optional[str] = None,
+        docker_network: Optional[str] = None,
+        log_file: Optional[str] = None,
+        skip_pull_image: Optional[bool] = None,
+        debug_ports: Optional[Tuple[int]] = None,
```

Comment:
Just wondering what is the difference between `Optional[Tuple[int]]` vs `Optional[List[int]]`?

Reply:
both of them have index access `debug_ports[0]` and tuple is immutable. 

I set it to tuple due to its docstring

> debug_ports tuple(int)
            Ports to bind the debugger to

Reply:
types only checked during linting. So even this code is valid:

```py
list: str = [] # assigning a list to a string-type variable.
```